spec/helpers/service_helper_spec.rb

```diff
@@ -114,19 +114,31 @@
 
     before do
       allow(TradeTariffFrontend::ServiceChooser).to receive(:enabled?).and_return(service_choosing_enabled)
+      assign(:enable_service_switch_banner_in_action, true)
+      allow(helper).to receive(:request).and_return(request)
     end
 
     context 'when on xi sections page' do
       let(:request) { double('request', filtered_path: '/xi/sections') }
 
       it 'returns the full banner that allows users to toggle between the services' do
-        expect(service_switch_banner).to include(t("service_banner.big.#{choice}", link: switch_service_link))
+        expect(helper.service_switch_banner).to include(t("service_banner.big.#{choice}", link: helper.switch_service_link))
```

Comment:
If you don't invoke the helper and just rely on method_missing to send the message to the helper we don't invoke the assigns that scope the instance variables inside of the view. This was actually quite painful to debug and is clearly a bug in RSpec.